From 80a453453746559084608f1ee60f987452be9209 Mon Sep 17 00:00:00 2001
From: Matheus Castanho <msc@linux.ibm.com>
Date: Mon, 3 May 2021 11:10:43 -0300
Subject: [PATCH 1/1] gold: Use nullptr to initialize Section_id with null
 pointer

Starting with GCC commit d96db15967e 'libstdc++: Deprecate non-standard
std::pair constructors [PR 99957]', gold build started to fail because NULL
cannot be used to initialize an empty Section_id (Section_id(NULL,0))
anymore. This commit fixes that by replacing NULL with nullptr.
---
 gold/ChangeLog | 6 ++++++
 gold/gc.h      | 4 ++--
 2 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/gold/ChangeLog b/gold/ChangeLog
index 82dae337bb3..47f87f4559f 100644
--- a/gold/ChangeLog
+++ b/gold/ChangeLog
@@ -1,3 +1,9 @@
+2021-05-03  Matheus Castanho  <msc@linux.ibm.com>
+
+	PR gold/27815
+	* gc.h (gc_process_relocs): Use nullptr instead of NULL on
+	Section_id constructor.
+
 2021-03-30  Alan Modra  <amodra@gmail.com>
 
 	PR gold/27625
diff --git a/gold/gc.h b/gold/gc.h
index 2c20b44b6ed..bd5a7f22b55 100644
--- a/gold/gc.h
+++ b/gold/gc.h
@@ -247,7 +247,7 @@ gc_process_relocs(
 	      if (is_ordinary) 
 		(*secvec).push_back(Section_id(src_obj, dst_indx));
 	      else
-                (*secvec).push_back(Section_id(NULL, 0));
+                (*secvec).push_back(Section_id(nullptr, 0));
               // If the target of the relocation is an STT_SECTION symbol,
               // make a note of that by storing -1 in the symbol vector.
               if (lsym.get_st_type() == elfcpp::STT_SECTION)
@@ -329,7 +329,7 @@ gc_process_relocs(
               if (is_ordinary && dst_obj != NULL)
 		(*secvec).push_back(Section_id(dst_obj, dst_indx));
 	      else
-                (*secvec).push_back(Section_id(NULL, 0));
+                (*secvec).push_back(Section_id(nullptr, 0));
               (*symvec).push_back(gsym);
 	      (*addendvec).push_back(std::make_pair(
 					static_cast<long long>(symvalue),
-- 
2.30.2

